modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/AbstractReturnDelegate.java

```diff
@@ -68,6 +76,14 @@ protected AbstractReturnDelegate(ComponentModel componentModel,
                                    CursorProviderFactory cursorProviderFactory,
                                    MuleContext muleContext) {
     returnsListOfMessages = componentModel instanceof HasOutputModel && returnsListOfMessages((HasOutputModel) componentModel);
+
+    isMapType = componentModel instanceof HasOutputModel && isMap(((HasOutputModel) componentModel).getOutput().getType());
```

Comment:
follow the spirit of `returnsListOfMessages` and add the util method there. Also make the `instanceOf HasOutputModel` inside the methods